components/appmgt-extensions/org.wso2.carbon.appmgt.mdm.wso2mdm.direct/src/main/java/org/wso2/carbon/appmgt/mdm/wso2mdm/direct/internal/MDMComponent.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.appmgt.mdm.wso2mdm.direct.internal;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.ServiceRegistration;
+import org.osgi.service.component.ComponentContext;
+import org.wso2.carbon.appmgt.mdm.wso2mdm.direct.MDMOperationsImpl;
+import org.wso2.carbon.appmgt.mobile.interfaces.MDMOperations;
+
+/**
+ * @scr.component name="org.wso2.carbon.appmgt.mdm.wso2mdm.osgi.WSO2MDMComponent" immediate="true"
+ */
+
+public class MDMComponent {
```

Comment:
This class doesn't seem to have references to any declarative services. Therefore, an elegant option would be to make it extend the BundleActivator interface.